src/pages/Register/Register.tsx

```diff
@@ -1,56 +1,73 @@
 import { createUserWithEmailAndPassword, UserCredential } from "firebase/auth";
 import { capitalisedFirstLetters } from "../../utils/capitalisedFirstLetters";
-import { auth } from "../../firebase";
-import { db } from "../../firebase";
 import { ChangeEvent, FormEvent, useState } from "react";
 import { useNavigate } from "react-router-dom";
 import arrowLeft from "../../assets/images/arrow-left.png";
 import Button from "../../components/Button/Button";
-import { doc, setDoc } from "firebase/firestore";
 import "./Register.scss";
+import { db, storage, auth } from "../../firebase";
+import { doc, setDoc } from "firebase/firestore";
+import { ref, uploadBytes } from "firebase/storage";
 
 const emptyFormData = {
   firstName: "",
   lastName: "",
   email: "",
   password: "",
   confirmPassword: "",
+  bio: "",
+  img: "",
+  tribe: "",
 };
-
 type RegisterProps = {
   setUserUID: (userID: string) => void;
 };
-
 const Register = ({ setUserUID }: RegisterProps) => {
   const [formData, setFormData] = useState(emptyFormData);
   const [passwordMatchError, setPasswordMatchError] = useState<string>("");
+  const [missingFieldsError, setMissingFieldsError] = useState<string>("");
   const [showSecondForm, setShowSecondFrom] = useState<boolean>(false);
+  const [showUploadPrompt, setShowUploadPrompt] = useState<boolean>(false);
+  const [selectedFile, setSelectedFile] = useState<File | null>(null);
   const navigate = useNavigate();
+
   const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
     const { name, value } = event.target;
     setFormData({ ...formData, [name]: value });
   };
-
-  const handleNext = (event: React.MouseEvent<HTMLButtonElement>) => {
+  const handleNext = async (event: React.MouseEvent<HTMLButtonElement>) => {
     event.preventDefault();
-    if (formData.firstName && formData.lastName) {
+    if (
+      formData.firstName &&
+      formData.lastName &&
+      formData.bio &&
+      formData.tribe
+    ) {
       setShowSecondFrom(true);
     }
+    try {
```

Comment:
This can be simplified to 

```ts
    if (
      formData.firstName &&
      formData.lastName &&
      formData.bio &&
      formData.tribe
    ) {
      setShowSecondFrom(true);
    } else {
      setMissingFieldsError("Please fill all required fields.");
    }
  };
```